test/unit/forge/db/models/Device_spec.js

```diff
@@ -92,4 +92,350 @@ describe('Device model', function () {
+            const name = nameGenerator('Test Project')
+            const project = await Project.create({
+                name,
+                safeName: name.toLowerCase().replace(/[^a-z0-9]/g, '-'),
+                type: '',
+                url: '',
+                ApplicationId: applicationId,
+                TeamId: teamId,
+                ProjectTypeId: projectTypeId || null,
+                ProjectStackId: projectStackId || null,
+                ProjectTemplateId: projectTemplateId || null
+            })
+            return project
+        }
+        const newApplication = async (teamId) => {
+            const name = nameGenerator('Test Application')
+            const application = await Application.create({ name, TeamId: teamId })
+            return application
+        }
+        const newTeam = async (teamTypeId = 1) => {
+            const name = nameGenerator('Test Team')
+            const team = await Team.create({ name, TeamTypeId: teamTypeId })
+            return team
+        }
+        const newDevice = async (teamId, { targetSnapshotId, activeSnapshotId, applicationId, projectId, deviceGroupId } = {}) => {
+            const name = nameGenerator('Test Device')
+            const device = await Device.create({
+                name,
+                TeamId: teamId,
+                type: 'PI',
+                credentialSecret: 'abc',
+                state: 'active',
+                targetSnapshotId: targetSnapshotId || null,
+                activeSnapshotId: activeSnapshotId || null,
+                ApplicationId: applicationId || null,
+                ProjectId: projectId || null,
+                DeviceGroupId: deviceGroupId || null
+            })
+            return device
+        }
+        const newDeviceGroup = async (applicationId, targetSnapshotId) => {
+            const name = nameGenerator('Test Device Group')
+            const deviceGroup = await app.db.models.DeviceGroup.create({ name, ApplicationId: applicationId, targetSnapshotId })
+            return deviceGroup
+        }
+        const newSnapshot = async () => {
+            const name = nameGenerator('Test Snapshot')
+            const snapshot = await app.db.models.ProjectSnapshot.create({ name })
+            return snapshot
+        }
+
+        // TESTS
+
+        // RELATION: TeamId INTEGER REFERENCES Teams (id) ON DELETE SET NULL ON UPDATE CASCADE,
+        it('should delete Device on Team delete', async () => {
+            // Create an application and a project
+            const team = await newTeam()
+            const device = await newDevice(team.id)
+            const did = device.id
+
+            // ensure device has team
+            device.should.have.property('TeamId', team.id)
+
+            // Delete the team
+            await team.destroy()
+
+            // Reload to get updated data
+            const updated = await Device.findByPk(did)
+            should(updated).not.be.null()
```

Comment:
If the device should be deleted when the team is deleted, why is this not null?

Reply:
Ben, the relation here is `ON DELETE SET NULL` 

So if a row in the `Teams` table is deleted, the `TeamId` in the `Device` table will be set to NULL - therefore the `updated` `Device` should still exist - only the `TeamId` should be `NULL`ed (that test is on the next line down)